datafusion/optimizer/src/optimizer.rs

```diff
@@ -184,8 +184,9 @@ impl Optimizer {
             rules.push(Arc::new(FilterNullJoinKeys::default()));
         }
         rules.push(Arc::new(EliminateOuterJoin::new()));
-        rules.push(Arc::new(FilterPushDown::new()));
+        // Filter can't pushdown Limit, we should do PushDownFilter after LimitPushDown
```

Comment:
```suggestion
        // Filters can't be pushed down past Limits, we should do PushDownFilter after LimitPushDown
```

Is this the intention of running Filter pushdown after limit pushdown?